### PR TITLE
ICU-21900 Update ICU4C API reference docs for 71.1

### PR DESCRIPTION
**Pull request by sven-oly**

### CODE REVIEW

**Review by richgillam — APPROVED** [review]
Too big to review.  I spot-checked the changes, and they appear mechanical and reasonable, so I'm rubber-stamping.

File: apidoc/released/icu4c/search/variables_d.html [hunk 1]
```diff
@@ -10,14 +11,14 @@
 <div id="SRIndex">
 <div class="SRStatus" id="Loading">Loading...</div>
 <div id="SRResults"></div>
-<script type="text/javascript"><!--
```
Comment (srl295) [inline apidoc/released/icu4c/search/variables_d.html:13]: Is this an issue? @markusicu (next line is the search stuff)

**Review by srl295 — APPROVED** [review]
spot check lgtm

**Comment by markusicu** [comment]
This was the wrong set of files. Before a release, update "dev". For the release itself, update "released".

https://unicode-org.github.io/icu/processes/release/tasks/docs.html#build-api-documentations “Copy the zip file into your local copy of icu-docs into apidoc/released/icu4c (**or apidoc/dev/icu4c if not a release**)”

**Pull request merged**